src/main/java/com/mraof/minestuck/effects/SoporSicknessEffect.java

```diff
@@ -0,0 +1,47 @@
+package com.mraof.minestuck.effects;
+
+import com.mraof.minestuck.Minestuck;
+import net.minecraft.world.effect.MobEffect;
+import net.minecraft.world.effect.MobEffectCategory;
+import net.minecraft.world.effect.MobEffectInstance;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraftforge.event.entity.living.MobEffectEvent;
+import net.minecraftforge.eventbus.api.Event;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import net.minecraftforge.fml.common.Mod;
+
+@Mod.EventBusSubscriber(modid = Minestuck.MOD_ID, bus = Mod.EventBusSubscriber.Bus.FORGE)
+public class SoporSicknessEffect extends MobEffect
+{
+	protected SoporSicknessEffect()
+	{
+		super(MobEffectCategory.NEUTRAL, 0x47453d);
+	}
+	
+	@Override
+	public void applyEffectTick(LivingEntity target, int effectLevel)
+	{
+		if (target.getHealth() > 5.0F)
+		{
+			target.hurt(target.damageSources().generic(), 0.5F);
+		}
+	}
+	
+	@SubscribeEvent
+	public static void effectApplicabilityEvent(MobEffectEvent.Applicable event)
+	{
+		LivingEntity entity = event.getEntity();
+		MobEffect sopor_sickness = MSEffects.SOPOR_SICKNESS.get();
+		
+		if(entity.hasEffect(sopor_sickness))
+		{
+			event.setResult(Event.Result.DENY);
+		}
```

Comment:
Being able to negate potion effects outside of vanilla can cause large compatibility issues, even just within Minestuck the current setup prevents Creative Shock from taking effect. I can recommend the following, although it should be passed by Kirderf

```suggestion
		LivingEntity entity = event.getEntity();
		String effectName = event.getEffectInstance().toString();
		
		//prevents effect from working if sopor sickness is used and it is a vanilla effect
		if(entity.hasEffect(MSEffects.SOPOR_SICKNESS.get()) && effectName.contains("minecraft"))
		{
			event.setResult(Event.Result.DENY);
		}
```

Reply:
I don't think using and checking `toString()` is particularly great here. I'd instead suggest making a data tag to act as a whitelist.

Reply:
Is there any examples to work from?

Reply:
You could look at our `SCANNER_LOCATED` structure tag and try to do something similar, but for mob effects instead of structures.

Reply:
A tag for MobEffect has been added to resolve this. Any potion that wishes to be impacted by Sopor Sickness must exist in said whitelist tag for it to do so